modules/s1/main.tf

```diff
@@ -0,0 +1,570 @@
+  description = "store_incident_lambda_policy"
+
+  policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": [
+        "sqs:ReceiveMessage",
+        "sqs:DeleteMessage",
+        "sqs:GetQueueAttributes"
+      ],
+      "Effect": "Allow",
+      "Resource": "${aws_sqs_queue.incidents-queue.arn}"
+    },
+    {
+      "Action": [
+        "rds-data:ExecuteStatement",
+        "rds-data:RollbackTransaction",
+        "rds-data:CommitTransaction",
+        "rds-data:ExecuteSql",
+        "rds-data:BatchExecuteStatement",
+        "rds-data:BeginTransaction"
+      ],
+      "Effect": "Allow",
+      "Resource": "${aws_rds_cluster.incidents.arn}"
+    },
+    {
+      "Action": [
+        "logs:CreateLogGroup",
+        "logs:CreateLogStream",
+        "logs:PutLogEvents"
+      ],
+      "Effect": "Allow",
+      "Resource": "*"
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_iam_role" "store_incident_lambda_policy" {
+  name               = "store_incident_lambda_policy"
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": "sts:AssumeRole",
+      "Principal": {
+        "Service": "lambda.amazonaws.com"
+      },
+      "Effect": "Allow"
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_iam_role_policy_attachment" "store_incident_lambda_policy_attachment" {
+  role       = aws_iam_role.store_incident_lambda_policy.id
+  policy_arn = aws_iam_policy.store_incident_lambda_policy.arn
+}
+
+resource "aws_iam_policy" "read_incident_lambda_policy" {
+  name        = "store_incident_lambda_policy"
+  description = "store_incident_lambda_policy"
+
+  policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": [
+        "rds-data:ExecuteStatement",
+        "rds-data:RollbackTransaction",
+        "rds-data:CommitTransaction",
+        "rds-data:ExecuteSql",
+        "rds-data:BatchExecuteStatement",
+        "rds-data:BeginTransaction"
+      ],
+      "Effect": "Allow",
+      "Resource": "${aws_rds_cluster.incidents.arn}" TODO
+    },
+    {
+      "Action": [
+        "logs:CreateLogGroup",
+        "logs:CreateLogStream",
+        "logs:PutLogEvents"
+      ],
+      "Effect": "Allow",
+      "Resource": "*"
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_iam_role" "read_incident_lambda_policy" {
+  name               = "read_incident_lambda_policy"
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": "sts:AssumeRole",
+      "Principal": {
+        "Service": "lambda.amazonaws.com"
+      },
+      "Effect": "Allow"
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_iam_role_policy_attachment" "read_incident_lambda_policy_attachment" {
+  role       = aws_iam_role.read_incident_lambda_policy.id
+  policy_arn = aws_iam_policy.read_incident_lambda_policy.arn
+}
+
+### Backend S3
+resource "aws_s3_bucket" "ugt_lambda_states" {
+  bucket = join("-", [var.env_name, var.region, "lambda-states"])
```

Comment:
Guys, I'm curious about this one. Are we pushing all squads lambdas to same bucket? 

By checking S3(squad) module and resource "aws_s3_bucket", it follows same name from this PR. If we are not using same bucket for all squads, this will be a conflict since S3 bucket names are globally unique. We could add account id or some other identifier to the bucket name.

Reply:
Good catch. This project should live in a different account, so we shouldn’t reuse the same bucket. The naming collision we can avoid by using https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/s3_bucket#bucket_prefix.

Reply:
Although env name would be s1, I believe. In this case there wouldn’t be a clash.

Reply:
Each project lives on different accounts on AWS in order to avoid any issue and mitigate risks